src/buffer/out/Row.cpp

```diff
@@ -118,10 +209,16 @@ LineRendition ROW::GetLineRendition() const noexcept
     return _lineRendition;
 }
 
-uint16_t ROW::GetLineWidth() const noexcept
+// Returns the index 1 past the last (technically) valid column in the row.
+// The interplay between the old console and newer VT APIs which support line renditions is
+// still unclear so it might be necessary to add two kinds of this function in the future.
+// Console APIs treat the buffer as a large NxM matrix after all.
+til::CoordType ROW::GetReadableColumnCount() const noexcept
 {
-    const auto scale = _lineRendition != LineRendition::SingleWidth ? 1 : 0;
-    return _columnCount >> scale;
+    const til::CoordType columnCount = _columnCount;
+    const til::CoordType scale = _lineRendition != LineRendition::SingleWidth;
+    const til::CoordType padding = _doubleBytePadded;
+    return (columnCount - padding) >> scale;

```

Comment:
Yeah, I was really unsure in what order these need to be, and I still am a little.
Thinking about it some more, does it have to be like this?
```cpp
const auto padding = _doubleBytePadded << scale;
return (columnCount - padding) >> scale;
```

Reply:
That would be the same as...

```cpp
return (columnCount >> scale) - padding;
```

right?

Reply:
Not quite since the rounding is different. In your code it would subtract 2 no matter what. For instance, imagine padding is 2 and columnCount is 11. In my code snippet it would return 4 and with yours it would return 3. I think 4 is correct because 4*2 + (anything between 1 and 3 for the wide glyph padding) is 10.

Reply:
Good call.